### PR TITLE
chore: checkout last 5 commits

### PR DESCRIPTION
**Pull request by DanielRuf**

### CODE REVIEW

**Comment by wmhilton** [comment]
This maybe speeds up git clone from 1s to 0.9s. Or slows it down... from .8s to .9s? It's hard to tell without running a bunch and averaging. :( Regardless, pretty small dividends. I appreciate the effort though!

**Comment by DanielRuf** [comment]
Hm, good question. Might depend on a few other things too.
I guess I have to test and benchmark it in my fork.

**Comment by DanielRuf** [comment]
Also see https://docs.travis-ci.com/user/reference/overview/#Virtualisation-Environment-vs-Operating-System

**Comment by wmhilton** [comment]
Looks like it's running in ec2 (so the container based) environment.
If we could use a custom docker image with npm 5.7 we'd be golden.

Also, I have it test using a local Firefox, because for PRs the environment variables for Sauce aren't present. But maybe turning off the local Firefox test when the Sauce environment variables are present would speed things up by freeing up RAM?

**Comment by DanielRuf** [comment]
>  because for PRs the environment variables for Sauce aren't present

If I remember correctly we solved it for foundation-sites and a few other projects. Will have to check how we did it there.

> But maybe turning off the local Firefox test when the Sauce environment variables are present would speed things up by freeing up RAM?

Probably, yes. Actually never used a local Firefox instance (we use puppeteer in some projects) in CI setups.

**Comment by DanielRuf** [comment]
Unfortunately I do not know what the solution is now after the JWT addon was deprecated.

**Comment by wmhilton** [comment]
> Actually never used a local Firefox instance (we use puppeteer in some projects) in CI setups.

I think I was trying to use puppeteer / HeadlessChrome at one point, but it had issues working on Travis which is why I ended up with somewhat-headless Firefox.

**Comment by DanielRuf** [comment]
Hm, which exact issues? I can take a look. I've migrated phantomjs to puppeteer at zurb/foundation-sites and can help here too.

**Comment by wmhilton** [comment]
I think I was trying to use [ChromeHeadless](https://github.com/isomorphic-git/isomorphic-git/commit/6684e9b89b24ce45086d703ecc80c61893385759#diff-a2a3b7b0c9c3b4b93b4aebf4e3ec3cfb) using the Travis addon, which is [documented](https://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-the-Chrome-addon-in-the-headless-mode) but failed with [this error](https://github.com/travis-ci/travis-ci/issues/9024) so I [removed](https://github.com/isomorphic-git/isomorphic-git/commit/f363c05ab8bed1416e0c62bcf74a559cf9f8cc11#diff-a2a3b7b0c9c3b4b93b4aebf4e3ec3cfb) it.

[Searching through the repo](https://github.com/isomorphic-git/isomorphic-git/search?q=puppeteer&type=Issues) for puppeteer specifically, it doesn't look like I ever tried it specifically. But what's this? Puppeteer is mentioned in the release notes for Jest! `3b0959adocs: mention Jest Puppeteer Preset (#​5722)` (I don't always live tweet my browsing, but when I do, I try to be dramatic.) 

> +[Jest Puppeteer Preset](https://github.com/smooth-code/jest-puppeteer) provides
> +all required configuration to run your tests using Puppeteer.

Sweet. I was actually thinking of running the `karma` tests in puppeteer, but if we can trick `jest` into running them that works too.

Long story short... I've already rewritten the tests like 4 times (ava -> jest -> jasmine -> karma) and don't want to have to change them to make them work with puppeteer... if there's a karma-puppeteer... oh hang on I found a Medium article. Lemme try this brb.

**Comment by DanielRuf** [comment]
There is a tool to rewrite them almost automagically.

**Comment by DanielRuf** [comment]
Take a look at the draft of my current article (you can use Google Translate as it is in German): https://bitspeicher.blog/p/7c765cbe-7481-46ae-aa34-94a1b0e41311/

**Comment by DanielRuf** [comment]
> if there's a karma-puppeteer... oh hang on I found a Medium article. Lemme try this brb.

Right, there are many projest like jest-puppeteer and also config sets for karma afaik.

**Comment by wmhilton** [comment]
I'm trying a thing here: https://github.com/isomorphic-git/isomorphic-git/pull/182

**Comment by wmhilton** [comment]
I'm closing this since the amount of time it might save is smaller than the time it would take to benchmark it and see if it really saves any time.
